exec/java-exec/src/test/java/org/apache/drill/test/OperatorTestBuilder.java

```diff
@@ -0,0 +1,300 @@
+    CloseableRecordBatch testOperator = null;
+
+    try {
+      validate();
+      int expectedNumBatches = expectedNumBatchesOpt.orElse(expectedResults.size());
+      physicalOpUnitTestBase.mockOpContext(physicalOperator, initReservation, maxAllocation);
+
+      final BatchCreator<PhysicalOperator> opCreator = (BatchCreator<PhysicalOperator>) physicalOpUnitTestBase.opCreatorReg.getOperatorCreator(physicalOperator.getClass());
+      testOperator = opCreator.getBatch(physicalOpUnitTestBase.fragContext, physicalOperator, (List)upstreamBatches);
+
+      batchIterator: for (int batchIndex = 0;; batchIndex++) {
+        final RecordBatch.IterOutcome outcome = testOperator.next();
+
+        switch (outcome) {
+          case NONE:
+            if (!combineOutputBatches) {
+              Assert.assertEquals(expectedNumBatches, batchIndex);
+            }
+            // We are done iterating over batches. Now we need to compare them.
+            break batchIterator;
+          case OK_NEW_SCHEMA:
+            boolean skip = true;
+
+            try {
+              skip = testOperator.getContainer().getRecordCount() == 0;
+            } catch (IllegalStateException e) {
+              // We should skip this batch in this case. It means no data was included with the okay schema
+            } finally {
+              if (skip) {
+                batchIndex--;
+                break;
+              }
+            }
+          case OK:
+            if (!combineOutputBatches && batchIndex >= expectedNumBatches) {
+              testOperator.getContainer().clear();
+              Assert.fail("More batches received than expected.");
+            } else {
+              final boolean hasSelectionVector = testOperator.getSchema().getSelectionVectorMode().hasSelectionVector;
+              final VectorContainer container = testOperator.getContainer();
+
+              if (hasSelectionVector) {
+                actualResults.add(IndirectRowSet.fromSv2(testOperator.getContainer(), testOperator.getSelectionVector2()));
+              } else {
+                actualResults.add(DirectRowSet.fromContainer(container));
+              }
+              break;
+            }
+          default:
+            throw new UnsupportedOperationException("Can't handle this yet");
+        }
+      }
+
+      int actualTotalRows = actualResults.stream()
+        .map(rowSet -> rowSet.rowCount())
+        .reduce((Integer countA, Integer countB) -> countA + countB)
+        .orElse(0);
+
+      if (expectedResults.isEmpty()) {
+        Assert.assertEquals((int) expectedTotalRowsOpt.orElse(0), actualTotalRows);
+        // We are done, we don't have any expected result to compare
+        return;
+      }
+
+      if (combineOutputBatches) {
+        final RowSet expectedBatch = expectedResults.get(0);
+        final RowSet actualBatch = DirectRowSet.fromSchema(
+          physicalOpUnitTestBase.operatorFixture.allocator, expectedBatch.schema());
```

Comment:
shouldn't this be `actualResults.get(0).container().schema()` instead of `expectedBatch.schema()` ?

Reply:
Changed.